style/js/shield_text.js

```diff
@@ -69,7 +102,21 @@ export function layoutShieldText(text, padding, bounds, textLayoutFunc) {
   metrics = ctx.measureText(text);
   textHeight = metrics.actualBoundingBoxDescent;
 
-  var yBaseline = padTop + (availHeight - textHeight) / 2;
+  var yBaseline;
+
+  switch (textConstraint.valign) {
+    case VerticalAlignment.Middle:
+      yBaseline = padTop + (availHeight - textHeight) / 2;
+      break;
+    case VerticalAlignment.Top:
+      yBaseline = padTop;
+      break;
+    case VerticalAlignment.Bottom:
+      yBaseline = padTop + availHeight - textHeight;
+      break;
+    default:
+      return null; //Code error, should never happen
```

Comment:
This seems fragile.  Maybe it would make sense to throw an exception here and add error handling at the top level (the `missingIconLoader()` function.  This could log a helpful message to the console when this code error does happen.  It's hard to say what sort of error returning `null` here might produce elsewhere in the code, but a custom error message in that unlikely case would probably be clearer.